### PR TITLE
Expose hystrix.stream endpoint

### PR DESCRIPTION
**Pull request by aramirez-es**
**What?**
Basically this PR add **by default** an endpoint `GET /hystrix.stream` that exposes the Hystrix metrics as stream usable by both Hystrix Dashboard and Turbine (check out https://github.com/scm-spain/turbine-hystrix-dashboard for more information).

**How does it work?**
Using Ribbon in our micro services generates Hystrix Metrics (which we were not taking advantage of). That Hystrix metrics are readable in any point of our projects, so this PR exposes that metrics.

By using a Hystrix - RxNetty connector (https://github.com/Netflix/Hystrix/tree/master/hystrix-contrib/hystrix-rx-netty-metrics-stream), we have that endpoint that exposes metrics.

Router is now bound to a RouterProvider, which is in charge of check whether metrics must be exposed as stream (true by default but configurable via properties), if so, it returns `HystrixMetricsStreamHandler` using `RestRouterHandler` as delegated. Otherwise returns RestRouterHandler directly.

**Configuration**
By default this endpoint is exposed, but you can disable by adding next property to AppServer.properties:

```
com.scmspain.karyon.rest.property.exposeHystrixMetrics=false
```

**Releasing**
As I said before, it's not fully tested due to #21. So, please discuss this PR but do **not merge it yet**.
As this PR add a new feature, I propose to increase the minor version to **1.5**. Since it's fully backward-compatible and add a new feature.

Can you guys take a look, please?

### CODE REVIEW

File: src/test/java/scmspain/karyon/restrouter/KaryonRestRouterTest.java [hunk 1]
```diff
@@ -246,4 +251,33 @@ public void testEndpointWithQueryParamsReturnIntegerObs() throws Exception {
...
+    new HystrixObservableCommand<HttpClientResponse<ByteBuf>>(HystrixCommandGroupKey.Factory.asKey("request")) {
+      @Override
+      protected Observable<HttpClientResponse<ByteBuf>> construct() {
+        return RxNetty.createHttpClient("localhost", AppServer.KaryonRestRouterModuleImpl.DEFAULT_PORT)
+          .submit(HttpClientRequest.createGet("/example/1"));
+      }
+    }.construct().map(this::callToHystrixStreamEndpoint);
+  }
+
+  private String callToHystrixStreamEndpoint(HttpClientResponse<ByteBuf> any) throws RuntimeException {
```
Comment (victuxbb) [inline src/test/java/scmspain/karyon/restrouter/KaryonRestRouterTest.java:269]: this test finally works? (callToHystrixStreamEndpoint)

Comment (fiunchinho) [inline src/test/java/scmspain/karyon/restrouter/KaryonRestRouterTest.java:269]: When Alberto told me that FINALLY they found out the reason...

Comment (fiunchinho) [inline src/test/java/scmspain/karyon/restrouter/KaryonRestRouterTest.java:269]: Oh, I just saw that you guys deleted the whole test.

Comment (aramirez-es) [inline src/test/java/scmspain/karyon/restrouter/KaryonRestRouterTest.java:269]: I deleted the test as last chance before commit suicide xD
Just read carefully tests' documentation: https://github.com/scm-spain/karyon-rest-router/pull/25/files#diff-affe0e4f87e820c2f801be43cbddf613L68

Comment (aramirez-es) [inline src/test/java/scmspain/karyon/restrouter/KaryonRestRouterTest.java:269]: Anyway it's not related with this tests at all. RestRouterScannerTest change "mock" the property _com.scmspain.karyon.rest.property.packages_ and the state is not cleaned up later, so the routes are not loaded and KaryonRestRouterTest's endpoints returns 404.

**Comment by padilo** [comment]
This test class was refactor this way in #23 to check that is working properly when the property 'com.scmspain.karyon.rest.property.packages' isn't set.

Instead of deleting the whole test class please add the class annotation `@Ignore` to the class with a description, this should ignore the class test.

If I have some time I'll take a look to recover this test.

**Comment by aramirez-es** [comment]
I'm sorry guys, no more time spend here from my side. I'm closing this PR, is up to you guys to recover and merge these changes in the future.

**Pull request closed without merging**